parsons/newmode/newmode_v2.py

```diff
@@ -0,0 +1,237 @@
+                env variable set.
+            api_version: str
+                The api version to use. Defaults to v1.0
+        Returns:
+            NewMode Class
+        """
+        self.api_version = check_env.check("NEWMODE_API_VERSION", api_version)
+        self.base_url = API_URL
+        self.client_id = check_env.check("NEWMODE_API_CLIENT_ID", client_id)
+        self.__client_secret = check_env.check("NEWMODE_API_CLIENT_SECRET", client_secret)
+        self.headers = {"content-type": "application/json"}
+
+    def convert_to_table(self, data):
+        """Internal method to create a Parsons table from a data element."""
+        table = None
+        if type(data) is list:
+            table = Table(data)
+        else:
+            table = Table([data])
+
+        return table
+
+    def base_request(self, method, url, data=None, json=None, data_key=None, params={}):
+        response = None
+        response = self.client.request(
+            url=url, req_type=method, json=json, data=data, params=params
+        )
+        response.raise_for_status()
+        success_codes = [200, 201, 202, 204]
+        self.client.validate_response(response)
+        if response.status_code in success_codes:
+            if self.client.json_check(response):
+                response_json = response.json()
+                return response_json[data_key] if data_key else response_json
+            else:
+                return response.status_code
+        return response
+
+    def converted_request(
+        self,
+        endpoint,
+        method,
+        supports_version=True,
+        data=None,
+        json=None,
+        params={},
+        convert_to_table=True,
+        data_key=None,
+    ):
+        self.client = OAuth2APIConnector(
+            uri=self.base_url,
+            auto_refresh_url=API_AUTH_URL,
+            client_id=self.client_id,
+            client_secret=self.__client_secret,
+            headers=self.headers,
+            token_url=API_AUTH_URL,
+            grant_type="client_credentials",
+        )
+
+        url = f"{self.api_version}/{endpoint}" if supports_version else endpoint
+        response = self.base_request(
+            method=method,
+            url=url,
+            json=json,
+            data=data,
+            params=params,
+            data_key=data_key,
+        )
+        if not response:
+            logging.warning(f"Empty result returned from endpoint: {endpoint}")
+        if convert_to_table:
+            return self.convert_to_table(response)
+        else:
+            return response
+
+    def get_campaign(self, campaign_id, params={}):
+        """
+        Retrieve a specific campaign by ID.
+
+        In v2, a campaign is equivalent to Tools or Actions in V1.
+        `Args:`
+            campaign_id: str
+                The ID of the campaign to retrieve.
+            params: dict
```

Comment:
If this is easy/straightforward to do it would be great to include a link here to newmode's docs to show what params are allowed and how to format them (same for all methods that accept `params`)